lib/muontrap/port.ex

```diff
@@ -112,4 +121,12 @@ defmodule MuonTrap.Port do
 
   defp encode_acks_helper(full_acks, partial_acks),
     do: [:binary.copy(<<255>>, full_acks), partial_acks - 1]
+
+  defp maybe_start_timer(timeout) when is_integer(timeout),
+    do: Process.send_after(self(), :timeout, timeout)
+
+  defp maybe_start_timer(_), do: nil
+
+  defp maybe_stop_timer(nil), do: :ok
+  defp maybe_stop_timer(timer_ref), do: Process.cancel_timer(timer_ref, info: false)
```

Comment:
There's a small race condition if the timeout happens around the same time as the command completes. Here's an idea at removing the message from the mailbox if the timer wasn't cancelled. This uses the timeout_message from the other comment.

```suggestion
  defp maybe_stop_timer({timer_ref, timeout_message}) do 
    if not Process.cancel_timer(timer_ref, info: false) do
      receive do
        ^timeout_message -> :ok
      after
        0 -> :ok
      end
    end    
```